src/models/base.model.js

```diff
@@ -1,30 +1,100 @@
 'use strict'
 
+const ObjectPath = require('object-path')
 const LoggingService = require('../services/logging.service')
 const DynamicsDalService = require('../services/dynamicsDal.service')
+const Utilities = require('../utilities/utilities')
 
 module.exports = class BaseModel {
+  constructor (...args) {
+    const data = args.pop()
+    if (data) {
+      Object.assign(this, data)
+    }
+    Utilities.convertFromDynamics(this)
+  }
+
+  static selectedDynamicsFields (filter = () => true) {
+    return this.mapping()
+      .filter(filter)
+      .map(({dynamics}) => dynamics)
+  }
+
+  static dynamicsToModel (dynamicsData, filter = () => true) {
+    const modelData = {}
+    this.mapping()
+      .filter(filter)
+        // ignore default values as they will only be set when writing to dynamics
+      .filter(({defaultVal}) => !defaultVal)
+      .forEach(({field, dynamics}) => {
+        // set values in javascript objects by specifying a path eg 'dob.month'.
+        // if the path doesn't exist yet, it will be created.
+        ObjectPath.set(modelData, field, dynamicsData[dynamics])
+      })
+    return new this(modelData)
+  }
+
+  modelToDynamics (filter = () => true) {
+    const dynamicsData = {}
+    this.constructor.mapping()
+      .filter(filter)
+      // ignore readonly values as they will only be set when reading from dynamics
+      .filter(({readOnly}) => !readOnly)
+      .forEach(({field, dynamics, defaultVal, bind}) => {
+        if (bind) {
+          if (this[field]) {
+            dynamicsData[`${bind.id}@odata.bind`] = `${bind.entity}(${this[field]})`
+          }
+        } else {
+          if (this[field] === undefined && defaultVal !== undefined) {
+            dynamicsData[dynamics] = defaultVal
+          } else {
+            if (this[field] !== undefined || field !== 'id') {
+              dynamicsData[dynamics] = ObjectPath.get(this, field)
+            }
+          }
+        }
+      })
+    return dynamicsData
+  }
+
   toString () {
     // Class name
-    let returnValue = `${this.constructor.name}: {\n`
+    const className = this.constructor.name
```

Comment:
Why did this method need to be re-written?

Reply:
I changed the code to allow a toString of nested objects.